06.Lesson_ThinkingProperties/ThinkingPropertiesExercise/MeetingScheduler/README.md

```diff
@@ -0,0 +1,35 @@
+
+- ❓ Is there a way to check that a Meeting always starts as UNITIALIZED?
```

Comment:
you can write an invariant that states that all meetings are always initialized. This invariant will check that this property hold on initialization which might have the effect you are looking for. It will fail for other functions so you can add a `require false` in a preserved block and leave a comment that states that the purpose of the invariant is to check only the initialization state. you can also add a filtered block that filters the functions that can move a meeting out of the UNINITIALIZED state.

Reply:
I added a couple rules [here](https://github.com/AbhiGulati/Certora-Tutorials/pull/1/commits/ce993888b122dd83e8020acb21db3dad1e9d7c60) that demonstrate that meetings start uninitialized, and only `scheduleMeeting` can move them out of that state.

> you can also add a filtered block that filters the functions that can move a meeting out of the UNINITIALIZED state.
Was this in reference to the invariant that you mentioned above? Is it possible to use `filtered` with `invariant`s?

Reply:
Yeah you can add a filtered block for invariants. You add it the same way except it goes below the statement but above any preserved blocks you might have.